packages/valory/skills/check_stop_trading_abci/behaviours.py

```diff
@@ -69,6 +70,28 @@ def mech_request_count(self, mech_request_count: int) -> None:
         """Set the liveness period."""
         self._mech_request_count = mech_request_count
 
+    @property
+    def mech_marketplace_request_count(self) -> int:
+        """Get the mech marketplace request count."""
+        return self._mech_marketplace_request_count
+
+    @mech_marketplace_request_count.setter
+    def mech_marketplace_request_count(self, mech_marketplace_request_count: int) -> None:
+        """Set the mech marketplace request count."""
+        self._mech_marketplace_request_count = mech_marketplace_request_count
+
+    def _get_mech_marketplace_request_count(self) -> WaitableConditionType:
+        """Get the mech marketplace request count."""
+        status = yield from self.contract_interact(
+            contract_address=self.params.mech_marketplace_address,
+            contract_public_id=MechContract.contract_id,
+            contract_callable="get_requests_count",
+            data_key="requests_count",
+            placeholder=get_name(CheckStopTradingBehaviour.mech_marketplace_request_count),
+            address=self.synchronized_data.safe_contract_address,
+        )
+        return status
+
```

Comment:
Only the contract address and data key change. Maybe just move the condition from the `is_staking_kpi_met` to the `_get_mech_request_count` instead of introducing new attribute, properties, and method.